lib/smalloc.js

```diff
@@ -1,91 +1,27 @@
 'use strict';
 
-const smalloc = process.binding('smalloc');
-const kMaxLength = smalloc.kMaxLength;
-const kMinType = smalloc.kMinType;
-const kMaxType = smalloc.kMaxType;
-const util = require('util');
+const smalloc = require('internal/smalloc');
+const deprecate = require('util').deprecate;
 
-exports.alloc = alloc;
-exports.copyOnto = copyOnto;
-exports.dispose = dispose;
-exports.hasExternalData = hasExternalData;
+exports.alloc =
+    deprecate(smalloc.alloc, 'smalloc.alloc: Deprecated, use typed arrays');
+
+exports.copyOnto =
+    deprecate(smalloc.copyOnto,
+              'smalloc.copyOnto: Deprecated, use typed arrays');
+
+exports.dispose =
+    deprecate(smalloc.dispose,
+              'smalloc.dispose: Deprecated, use typed arrays');
+
+exports.hasExternalData =
+    deprecate(smalloc.hasExternalData,
+              'smalloc.hasExternalData: Deprecated, use typed arrays');
 
-// don't allow kMaxLength to accidentally be overwritten. it's a lot less
-// apparent when a primitive is accidentally changed.
 Object.defineProperty(exports, 'kMaxLength', {
-  enumerable: true, value: kMaxLength, writable: false
+  enumerable: true, value: smalloc.kMaxLength, writable: false
 });
 
 Object.defineProperty(exports, 'Types', {
-  enumerable: true, value: Object.freeze(smalloc.types), writable: false
+  enumerable: true, value: Object.freeze(smalloc.Types), writable: false
```

Comment:
It's because I moved the file, then added it again.  In the original, `.types` comes from the binding object and is exported as `smalloc.Types`.  Here, I'm re-exporting the latter.